src/app/features/asset-list/bitcoin/brc20-token-asset-list/brc20-token-asset-list.tsx

```diff
@@ -22,14 +24,29 @@ interface Brc20TokenAssetDetails {
 interface Brc20TokenAssetListProps {
   tokens: Brc20TokenAssetDetails[];
   variant?: AssetListVariant;
+  hasTokenSetter?: React.Dispatch<React.SetStateAction<boolean>>;
+  rightElementVariant: RightElementVariant;
 }
-export function Brc20TokenAssetList({ tokens, variant }: Brc20TokenAssetListProps) {
+export function Brc20TokenAssetList({
+  tokens,
+  variant,
+  hasTokenSetter,
+  rightElementVariant,
+}: Brc20TokenAssetListProps) {
   const navigate = useNavigate();
   const { balance, isLoading } = useCurrentBtcCryptoAssetBalanceNativeSegwit();
 
+  useEffect(() => {
+    if (hasTokenSetter && tokens.length) hasTokenSetter(true);
+  }, [tokens.length, hasTokenSetter]);
+
+  if (!tokens.length) return null;
+
   const hasPositiveBtcBalanceForFees =
     variant === 'interactive' && balance.availableBalance.amount.isGreaterThan(0);
 
+  const icon = <Brc20AvatarIcon />;
```

Comment:
I don't think this gives much benefit over just passing `<Brc20AvatarIcon />` directly